electrum/gui/kivy/main_window.py

```diff
@@ -1202,6 +1202,11 @@ def on_fee(self, event, *arg):
         self.set_fee_status()
 
     def protected(self, msg, f, args):
+        tx_btc = args[0].amount_msat/1000
```

Comment:
This function, `protected`, is used as an indirection to call other methods `f` when the user should be prompted for their PIN code. It is not only used when paying an invoice, but also when e.g. displaying their seed or exporint a private key or deleting the wallet.
With the app in kivy open, e.g. try tapping the wallet name in the top left corner, and then tap your seed near the middle of the screen ("Tap to show"). That call is going through `protected`.

I suggest you instead modify the msg argument being passed to `protected` here:
https://github.com/spesmilo/electrum/blob/5891e039b1db74c0eed14fbf9b9f4e61d9f3546c/electrum/gui/kivy/uix/screens.py#L344